docs/token-contracts/fa12/4-fa12-archetype.md

```diff
@@ -0,0 +1,447 @@
+---
+id: 4-fa12-archetype
+title: Deploy a Verified Implementation of FA1.2 Contract
+sidebar_label: FA1.2 Archetype
+---
+
+## Introduction
+This article presents the procedure to deploy a verified implementation of the FA1.2 contract.
+
+[Program verification](https://en.wikipedia.org/wiki/Formal_verification) is a technique to mathematically prove that a program has the required properties. It provides a high level of confidence that the contract executes correctly according to its specification.
+
+The formal properties of the FA1.2 contract are presented [below](#formal-properties).
+
+The contract and its formal specification are written in the [Archetype language](https://archetype-lang.org/), a high-level domain-specific language dedicated to the development of verified smart contracts for the Tezos blockchain.
+
+For verification purposes, Archetype translates the contract to the Why3 language; [Why3](http://why3.lri.fr/) is a generic program verification platform; it generates proof obligations and calls [SMT](https://en.wikipedia.org/wiki/Satisfiability_modulo_theories) solvers to solve them.
+
+## Generate Michelson
+The Archetype source code is available [here](https://github.com/edukera/archetype-lang/blob/0a5ad0832709ac102a14534f22d4f94cb185866d/contracts/fa12.arl).
+
+> SHA256 (fa12.arl) = d38260e5f55e630880ecd5d5ed32eb91988d2f34b43a7c031275d8f8a8715184
+
+The following command generates the Michelson version of the contract in the `fa12.tz` file:
+```
+$ archetype fa12.arl > fa12.tz
+```
+
+> Install [Archetype](https://docs.archetype-lang.org/getting-started-1#installation) (minimum required version: 1.2.0).
+
+## Originate contract
+
+This section presents how to originate (deploy) the contract on the Tezos testnet with the Tezos client. It assumes you already have an activated account (if not follow this procedure).
+
+The following command generates the Micheline syntax for the initial storage:
+
+```
+$ archetype --set-caller-init tz1dEezCSqMVsmfnaAZS2rBJZfdFCZEyA7DP -t michelson-storage fa12.arl
+(Pair {  } { Elt "tz1dEezCSqMVsmfnaAZS2rBJZfdFCZEyA7DP" 10000000 })
+```
+
+The address passed as the initial caller (here `tz1dEezCSqMVsmfnaAZS2rBJZfdFCZEyA7DP`) is the address that possesses the total number of tokens (here 100000000). The generated initial storage is used to generate the origination command.
+
+The following command originates the contract:
+```
+$ tezos-client -S -A testnet-tezos.giganode.io -P 443 originate contract fa12 transferring 0 from alice running fa12.tz --init '(Pair {  } { Elt "tz1dEezCSqMVsmfnaAZS2rBJZfdFCZEyA7DP" 10000000 })' --burn-cap 0 --force -D
+```
```

Comment:
For homogeneity, we should point users towards configuring their client: https://assets.tqtezos.com/docs/setup/1-tezos-client/ (and https://assets.tqtezos.com/docs/setup/2-sandbox), instead of placing `-A` and `-P` options throughout the tutorials.

Reply:
Done. 
* removed `-S` `-A` `-P` parameters
* now pointing to `tezos-client` and `tezos-sanbbox` pages